openwisp_controller/connection/api/serializers.py

```diff
@@ -31,6 +31,21 @@ class CommandSerializer(ValidatedDeviceFieldSerializer):
         pk_field=serializers.UUIDField(format='hex_verbose'),
     )
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # show only connections and command types available for the device
+        if (
+            device := Device.objects.filter(pk=self.context.get('device_id'))
+            .only('organization_id', 'id')
+            .first()
+        ):
+            self.fields['connection'].queryset = DeviceConnection.objects.filter(
+                device=device
+            )
```

Comment:
What about something like:
```suggestion
        device_id = self.context.get('device_id')
        if device_id:
            self.fields['connection'].queryset = DeviceConnection.objects.filter(
                device_id=self.context.get('device_id')
            )```